File: C#/TestProject2(PageObj)/UnitTestProject1/Elements/SearchBar.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using OpenQA.Selenium;
using OpenQA.Selenium.Support.PageObjects;

namespace DoubleGis.Erm.UnitTestProject1.Elements
{
    public class SearchBar
    {
        private readonly IWebElement currentElement;

        public bool IsDisplayed => currentElement.Displayed;
        public bool IsEnabled => currentElement.Enabled;

        public IWebElement TextField => currentElement.FindElement(By.XPath(".//*[contains(@class, 'search-bar__field')]"));

        public IWebElement SearchButton => currentElement.FindElement(By.XPath(".//*[contains(@class, 'search-bar__button')]"));

        public SearchBar(IWebElement searchBarElement)
        {
            currentElement = searchBarElement;
        }

        public string GetPlaceholderText()
        {
            return TextField.GetAttribute("placeholder");
        }

        public void Search(string searchText) { }
    }
}

File: Python/uitests/elements/search_bar.py
__author__ = 'neliko'
from elements.base_component import BaseComponent


class SearchBar(BaseComponent):

    @property
    def text_field(self):
        return self.driver.find_element_by_xpath("//*[contains(@class, 'searchbar__field')]")
    @property
    def search_button(self):
        return self.driver.find_element_by_xpath("//*[contains(@class, 'searchbar__button')]")
    @property
    def placeholder(self):
        return self.text_field.get_attribute("placeholder")
